tests/zfs-tests/tests/functional/cache/cache_010_pos.ksh

```diff
@@ -56,29 +56,29 @@ function cleanup_testenv
 	fi
 }
 
-log_assert "Cache device can only be block devices."
+log_assert "Cache device can be files, character devices or block devices."
 log_onexit cleanup_testenv
 
 TESTVOL=testvol1$$
 dsk1=${DISKS%% *}
 log_must zpool create $TESTPOOL ${DISKS#$dsk1}
-
 # Add nomal ${DEV_RDSKDIR} device
 log_must zpool add $TESTPOOL cache \
     ${DEV_RDSKDIR}/${dsk1}${SLICE_PREFIX}${SLICE0}
 log_must verify_cache_device $TESTPOOL $dsk1 'ONLINE'
 
 # Add normal file
-log_mustnot zpool add $TESTPOOL cache $VDEV2
+log_must zpool add $TESTPOOL cache $VDEV2
 
 # Add /dev/rlofi device (allowed under Linux)
 if is_linux; then
 	lofidev=$(losetup -f)
 	lofidev=${lofidev##*/}
+	log_must zpool remove $TESTPOOL ${VDEV2%% *}
```

Comment:
This differs from the original test file, but when I run the test suite without it zfs complains that the device is already part of the pool.

Reply:
This would be expected since previously the test expected the file cache device to fail.